purchase_order_price_recalculation/models/purchase_order.py

```diff
@@ -0,0 +1,24 @@
+# Copyright 2022 Tecnativa - Víctor Martínez
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+from odoo import api, fields, models
+
+
+class PurchaseOrder(models.Model):
+    _inherit = "purchase.order"
+
+    show_update_line_info = fields.Boolean(store=False)
+
+    @api.onchange("partner_id")
+    def _onchange_partner_id(self):
+        self.show_update_line_info = (
+            self.partner_id
+            and self._origin.partner_id
+            and self._origin.partner_id != self.partner_id
```

Comment:
What about my suggestion of always marking this as True when changed?

Reply:
In my opinion, it should not be set to True in the onchange because that would cause the button to be displayed when a record is created (unnecessary).